nilearn/reporting/tests/test_html_report.py

```diff
@@ -13,53 +13,111 @@
-    # Dummy 3D data
+@pytest.fixture
+def data_img_3d():
+    """Dummy 3D data for testing."""
     data = np.zeros((9, 9, 9))
     data[3:-3, 3:-3, 3:-3] = 10
-    data_img_3d = Nifti1Image(data, np.eye(4))
+    return Nifti1Image(data, np.eye(4))
 
-    # Dummy mask
-    mask = np.zeros((9, 9, 9))
-    mask[4:-4, 4:-4, 4:-4] = True
-    mask_img_3d = Nifti1Image(data, np.eye(4))
 
-    # test .fit method
-    mask = input_data.NiftiMasker()
-    mask.fit(data_img_3d)
-    html = mask.generate_report()
+@pytest.fixture
+def mask():
+    """Dummy mask for testing."""
+    data = np.zeros((10, 10, 10), dtype=int)
+    data[3:7, 3:7, 3:7] = 1
+    return Nifti1Image(data, np.eye(4))
+
+
+@pytest.fixture
+def input_parameters(masker_class, data_img_3d):
+    n_regions = 9
+    shape = (13, 11, 12)
+    affine = np.diag([2, 2, 2, 1])
+    if masker_class == NiftiMasker:
+        return {"mask_img": data_img_3d}
+    if masker_class == NiftiLabelsMasker:
+        label_img = generate_labeled_regions(
+            shape, n_regions=n_regions, affine=affine
+        )
+        return {"labels_img": label_img}
+    elif masker_class == NiftiMapsMasker:
+        label_img, _ = generate_maps(
+            shape, n_regions=n_regions, affine=affine
+        )
+        return {"maps_img": label_img}
+
+
+@pytest.mark.parametrize("masker_class",
+                         [NiftiMasker, NiftiLabelsMasker, NiftiMapsMasker])
+def test_report_empty_fit(masker_class, input_parameters):
+    """Test minimal report generation."""
+    masker = masker_class(**input_parameters)
+    masker.fit()
+    _check_html(masker.generate_report())
+
+
+@pytest.mark.parametrize("masker_class",
+                         [NiftiMasker, NiftiLabelsMasker, NiftiMapsMasker])
+def test_empty_report(masker_class, input_parameters):
+    """Test with reports set to False."""
+    masker = masker_class(**input_parameters, reports=False)
+    masker.fit()
+    assert masker._reporting_data is None
+    assert masker._reporting() == [None]
+    with pytest.warns(UserWarning,
+                      match=("Report generation not enabled ! "
+                             "No visual outputs will be created.")):
+        masker.generate_report()
+
+
+@pytest.mark.parametrize("masker_class", [NiftiMasker, NiftiLabelsMasker])
+def test_reports_after_fit_3d_data(masker_class, input_parameters, data_img_3d):  # noqa
+    """Tests report generation after fitting on 3D data."""
+    masker = masker_class(**input_parameters)
+    masker.fit(data_img_3d)
+    html = masker.generate_report()
     _check_html(html)
 
-    # check providing mask to init
-    masker = input_data.NiftiMasker(mask_img=mask_img_3d)
+
+@pytest.mark.parametrize("masker_class", [NiftiMasker, NiftiLabelsMasker])
+def test_reports_after_fit_3d_data_with_mask(masker_class, input_parameters, data_img_3d, mask):  # noqa
+    """Tests report generation after fitting on 3D data with mask_img."""
+    if "mask_img" in input_parameters:
```

Comment:
Sorry if I miss something obvious, but you're going to parametrize the test in order to have input parameters of various kinds ?

Reply:
I'm still unsure. I initially thought that parametrizing the tests would be a good idea, but it might not be as straightforward as I thought. I will try to balance code factorization and readability and see how it goes.
For now, tests are still broken, but the rendering of the report in the example gallery is much better. See [here](https://8715-1235740-gh.circle-artifacts.com/0/dev/auto_examples/03_connectivity/plot_probabilistic_atlas_extraction.html#sphx-glr-auto-examples-03-connectivity-plot-probabilistic-atlas-extraction-py) for example.